Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -843,7 +909,38 @@ static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<Character
     if (*buffer != ']')
         return std::nullopt;
     buffer.advance();
-    return CalendarRecord { WTFMove(result) };
+
+    return CalendarRecord { isCritical, false, WTFMove(result) };
+}
+
+template<typename CharacterType>
+static std::optional<Vector<CalendarRecord>>
+parseCalendar(StringParsingBuffer<CharacterType>& buffer)
+{
+// https://tc39.es/proposal-temporal/#prod-Annotations
+//  Annotations :::
+//      Annotation Annotations[opt]
+
+    Vector<CalendarRecord> result;
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    bool calendarWasCritical = false;
+    while (canBeCalendar(buffer)) {
```

Comment:
If `canBeCalendar(buffer)` is `false`, it should return `std::nullopt` instead of an empty vector? That way we  can remove the size check on the caller